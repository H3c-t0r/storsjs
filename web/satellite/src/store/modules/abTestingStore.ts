// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

import { reactive } from 'vue';
import { defineStore } from 'pinia';

import { MetaUtils } from '@/utils/meta';
import { ABHitAction, ABTestApi, ABTestValues } from '@/types/abtesting';
import { ABHttpApi } from '@/api/abtesting';

export class ABTestingState {
    public abTestValues = new ABTestValues();
    public abTestingEnabled = MetaUtils.getMetaContent('ab-testing-enabled') === 'true';
    public abTestingInitialized = false;
}

export const useABTestingStore = defineStore('abTesting', () => {
    const state = reactive<ABTestingState>(new ABTestingState());

    const api: ABTestApi = new ABHttpApi();

    async function fetchValues(): Promise<ABTestValues> {
        if (!state.abTestingEnabled) return state.abTestValues;

        const values = await api.fetchABTestValues();

        state.abTestValues = values;
        state.abTestingInitialized = true;

        return values;
    }

    async function hit(action: ABHitAction): Promise<void> {
        if (!state.abTestingEnabled) return;
        if (!state.abTestingInitialized) {
            await fetchValues();
        }
        switch (action) {
        case ABHitAction.UPGRADE_ACCOUNT_CLICKED:
            api.sendHit(ABHitAction.UPGRADE_ACCOUNT_CLICKED).catch(_ => {});
            break;
        }
    }

    function reset(): void {
        state.abTestValues = new ABTestValues();
        state.abTestingInitialized = false;
    }

    return {
        fetchValues,
        hit,
        reset,
    };
});
