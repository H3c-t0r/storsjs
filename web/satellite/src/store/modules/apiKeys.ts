// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/store';
import { ApiKey, ApiKeyCursor, ApiKeyOrderBy, ApiKeysApi, ApiKeysPage } from '@/types/apiKeys';
import { SortDirection } from '@/types/common';

export const API_KEYS_MUTATIONS = {
    SET_PAGE: 'setApiKeys',
    TOGGLE_SELECTION: 'toggleApiKeysSelection',
    CLEAR_SELECTION: 'clearApiKeysSelection',
    CLEAR: 'clearApiKeys',
    CHANGE_SORT_ORDER: 'changeApiKeysSortOrder',
    CHANGE_SORT_ORDER_DIRECTION: 'changeApiKeysSortOrderDirection',
    SET_SEARCH_QUERY: 'setApiKeysSearchQuery',
    SET_PAGE_NUMBER: 'setApiKeysPage',
};

const {
    SET_PAGE,
    TOGGLE_SELECTION,
    CLEAR_SELECTION,
    CLEAR,
    CHANGE_SORT_ORDER,
    CHANGE_SORT_ORDER_DIRECTION,
    SET_SEARCH_QUERY,
    SET_PAGE_NUMBER,
} = API_KEYS_MUTATIONS;

class ApiKeysState {
    public cursor: ApiKeyCursor = new ApiKeyCursor();
    public page: ApiKeysPage = new ApiKeysPage();
}

/**
 * creates apiKeys module with all dependencies
 *
 * @param api - apiKeys api
 */
export function makeApiKeysModule(api: ApiKeysApi): StoreModule<ApiKeysState> {
    return {
        state: new ApiKeysState(),
        mutations: {
            [SET_PAGE](state: ApiKeysState, page: ApiKeysPage) {
                state.page = page;
            },
            [SET_PAGE_NUMBER](state: ApiKeysState, pageNumber: number) {
                state.cursor.page = pageNumber;
            },
            [SET_SEARCH_QUERY](state: ApiKeysState, search: string) {
                state.cursor.search = search;
            },
            [CHANGE_SORT_ORDER](state: ApiKeysState, order: ApiKeyOrderBy) {
                state.cursor.order = order;
            },
            [CHANGE_SORT_ORDER_DIRECTION](state: ApiKeysState, direction: SortDirection) {
                state.cursor.orderDirection = direction;
            },
            [TOGGLE_SELECTION](state: ApiKeysState, apiKeyID: string) {
                state.page.apiKeys = state.page.apiKeys.map((apiKey: ApiKey) => {
                    if (apiKey.id === apiKeyID) {
                        apiKey.isSelected = !apiKey.isSelected;
                    }

                    return apiKey;
                });
            },
            [CLEAR_SELECTION](state: ApiKeysState) {
                state.page.apiKeys = state.page.apiKeys.map((apiKey: ApiKey) => {
                    apiKey.isSelected = false;

                    return apiKey;
                });
            },
            [CLEAR](state: ApiKeysState) {
                state.cursor = new ApiKeyCursor();
                state.page = new ApiKeysPage();
            },
        },
        actions: {
            fetchApiKeys: async function ({commit, rootGetters, state}, pageNumber: number): Promise<ApiKeysPage> {
                const projectId = rootGetters.selectedProject.id;
                commit(SET_PAGE_NUMBER, pageNumber);

                const apiKeys = await api.get(projectId, state.cursor);
                commit(SET_PAGE, apiKeys);

                return apiKeys;
            },
            createApiKey: async function ({commit, rootGetters}: any, name: string): Promise<ApiKey> {
                const projectId = rootGetters.selectedProject.id;

                const apiKey = await api.create(projectId, name);

                return apiKey;
            },
            deleteApiKey: async function({commit}: any, ids: string[]): Promise<void> {
                return await api.delete(ids);
            },
            setApiKeysSearchQuery: function ({commit}, search: string) {
                commit(SET_SEARCH_QUERY, search);
            },
            setApiKeysSortingBy: function ({commit}, order: ApiKeyOrderBy) {
                commit(CHANGE_SORT_ORDER, order);
            },
            setApiKeysSortingDirection: function ({commit}, direction: SortDirection) {
                commit(CHANGE_SORT_ORDER_DIRECTION, direction);
            },
            toggleApiKeySelection: function ({commit}, apiKeyID: string): void {
                commit(TOGGLE_SELECTION, apiKeyID);
            },
            clearApiKeySelection: function ({commit}): void {
                commit(CLEAR_SELECTION);
            },
            clearApiKeys: function ({commit}): void {
                commit(CLEAR);
            },
        },
        getters: {
            selectedApiKeys: (state: ApiKeysState) => state.page.apiKeys.filter((key: ApiKey) => key.isSelected),
            apiKeys: function (state: ApiKeysState): ApiKey[] {
                return state.page.apiKeys;
            },
        },
    };
}
