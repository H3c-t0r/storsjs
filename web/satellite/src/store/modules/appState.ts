// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

import { APP_STATE_MUTATIONS } from '../mutationConstants';

export const appStateModule = {
	state: {
		// Object that contains all states of views
		currentAppState: {
            isAddTeamMembersPopupShown: false
        },
	},
	mutations: {
		// Mutation changing add team members popup state
		[APP_STATE_MUTATIONS.SET_ADD_TEAMMEMBER_POPUP_STATE](state: any, isShown: boolean): void {
            state.currentAppState.isAddTeamMembersPopupShown = isShown;
		},
		
	},
	actions: {
		// Commits muttation for changing add team members popup state
		setAddTeamMembersPopup: function ({commit}: any, isShown: boolean): void {
			commit(APP_STATE_MUTATIONS.SET_ADD_TEAMMEMBER_POPUP_STATE, isShown);
		},
	},
};
