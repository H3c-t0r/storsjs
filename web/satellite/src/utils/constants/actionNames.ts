// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const NOTIFICATION_ACTIONS = {
    SUCCESS: 'success',
    ERROR: 'error',
    NOTIFY: 'notify',
    WARNING: 'WARNING',
    DELETE: 'deleteNotification',
    PAUSE: 'pauseNotification',
    RESUME: 'resumeNotification',
    CLEAR: 'clearNotifications',
};
