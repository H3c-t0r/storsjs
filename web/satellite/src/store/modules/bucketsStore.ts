// Copyright (C) 2022 Storj Labs, Inc.
// See LICENSE for copying information.

import { defineStore } from 'pinia';
import { reactive } from 'vue';

import { Bucket, BucketCursor, BucketPage, BucketsApi } from '@/types/buckets';
import { BucketsApiGql } from '@/api/buckets';
import { useProjectsStore } from '@/store/modules/projectsStore';

const BUCKETS_PAGE_LIMIT = 7;
const FIRST_PAGE = 1;

export class BucketsState {
    public allBucketNames: string[] = [];
    public cursor: BucketCursor = { limit: BUCKETS_PAGE_LIMIT, search: '', page: FIRST_PAGE };
    public page: BucketPage = { buckets: new Array<Bucket>(), currentPage: 1, pageCount: 1, offset: 0, limit: BUCKETS_PAGE_LIMIT, search: '', totalCount: 0 };
}

export const useBucketsStore = defineStore('buckets', () => {
    const state = reactive<BucketsState>(new BucketsState());

    const api: BucketsApi = new BucketsApiGql();

    function setBucketsSearch(search: string): void {
        state.cursor.search = search;
    }

    function clearBucketsState(): void {
        state.allBucketNames = [];
        state.cursor = new BucketCursor('', BUCKETS_PAGE_LIMIT, FIRST_PAGE);
        state.page = new BucketPage([], '', BUCKETS_PAGE_LIMIT, 0, 1, 1, 0);
    }

    async function fetchBuckets(page: number): Promise<void> {
        const { projectsState } = useProjectsStore();
        const projectID = projectsState.selectedProject.id;
        const before = new Date();
        state.cursor.page = page;

        state.page = await api.get(projectID, before, state.cursor);
    }

    async function fetchAllBucketsNames(): Promise<void> {
        const { projectsState } = useProjectsStore();
        const projectID = projectsState.selectedProject.id;

        state.allBucketNames = await api.getAllBucketNames(projectID);
    }

    return {
        bucketsState: state,
        setBucketsSearch,
        clearBucketsState,
        fetchBuckets,
        fetchAllBucketsNames,
    };
});
