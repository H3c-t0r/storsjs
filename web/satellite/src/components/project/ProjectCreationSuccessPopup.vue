// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div v-if="isPopupShown" class="project-creation-success-popup-container">
        <div class="project-creation-success-popup" id='successfulProjectCreationPopup'>
            <div class="project-creation-success-popup__info-panel-container">
                <ProjectCreationSuccessIcon/>
            </div>
            <div class="project-creation-success-popup__form-container">
                <h2 class="project-creation-success-popup__form-container__main-label-text">Congrats!</h2>
                <p class="project-creation-success-popup__form-container__confirmation-text">You just created your project. Next, we recommend you create your first API Key for this project. API Keys allow developers to manage their projects and build applications on top of the Storj network through our
                    <a class="project-creation-success-popup__form-container__confirmation-text__link" href="https://github.com/storj/storj/wiki/Uplink-CLI" target="_blank">Uplink CLI.</a>
                </p>
                <div class="project-creation-success-popup__form-container__button-container">
                    <VButton
                        label="I will do it later"
                        width="214px"
                        height="50px"
                        :on-press="onCloseClick"
                        is-white="true"
                    />
                    <VButton
                        label="Create first API Key"
                        width="214px"
                        height="50px"
                        :on-press="onCreateAPIKeyClick"
                    />
                </div>
            </div>
            <div class="project-creation-success-popup__close-cross-container" @click="onCloseClick">
                <CloseCrossIcon/>
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VButton from '@/components/common/VButton.vue';

import CloseCrossIcon from '@/../static/images/common/closeCross.svg';
import ProjectCreationSuccessIcon from '@/../static/images/project/projectCreationSuccess.svg';

import { RouteConfig } from '@/router';
import { APP_STATE_ACTIONS } from '@/utils/constants/actionNames';

@Component({
    components: {
        VButton,
        ProjectCreationSuccessIcon,
        CloseCrossIcon,
    },
})
export default class ProjectCreationSuccessPopup extends Vue {
    public onCloseClick(): void {
        this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_SUCCESSFUL_PROJECT_CREATION_POPUP);
    }

    public onCreateAPIKeyClick(): void {
        this.$router.push(RouteConfig.ApiKeys.path);
        this.onCloseClick();
    }

    public get isPopupShown(): boolean {
        return this.$store.state.appStateModule.appState.isSuccessfulProjectCreationPopupShown;
    }
}
</script>

<style scoped lang="scss">
    .project-creation-success-popup-container {
        position: fixed;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        background-color: rgba(134, 134, 148, 0.4);
        z-index: 1000;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .project-creation-success-popup {
        width: 100%;
        max-width: 845px;
        background-color: #fff;
        border-radius: 6px;
        display: flex;
        flex-direction: row;
        align-items: flex-start;
        position: relative;
        justify-content: center;
        padding: 80px 100px 80px 50px;

        &__info-panel-container {
            display: flex;
            flex-direction: column;
            justify-content: flex-start;
            align-items: center;
            margin-right: 100px;
            margin-top: 20px;
        }

        &__form-container {
            width: 100%;
            max-width: 440px;
            margin-top: 10px;

            &__main-label-text {
                font-family: 'font_bold', sans-serif;
                font-size: 32px;
                line-height: 39px;
                color: #384b65;
                margin: 0;
            }

            &__button-container {
                width: 100%;
                display: flex;
                flex-direction: row;
                justify-content: space-between;
                align-items: center;
                margin-top: 40px;
            }

            &__confirmation-text {
                font-family: 'font_medium', sans-serif;
                font-size: 16px;
                line-height: 21px;
                color: #354049;
                padding: 27px 0 0 0;
                margin: 0;

                &__link {
                    font-family: 'font_bold', sans-serif;
                    color: #2683ff;
                }
            }
        }

        &__close-cross-container {
            display: flex;
            justify-content: center;
            align-items: center;
            position: absolute;
            right: 30px;
            top: 40px;
            height: 24px;
            width: 24px;
            cursor: pointer;

            &:hover .close-cross-svg-path {
                fill: #2683ff;
            }
        }
    }

    @media screen and (max-width: 720px) {

        .project-creation-success-popup {

            &__info-panel-container {
                display: none;
            }

            &__form-container {

                &__button-container {
                    width: 100%;
                }
            }
        }
    }
</style>
