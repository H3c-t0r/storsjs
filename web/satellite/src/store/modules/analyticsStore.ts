// Copyright (C) 2022 Storj Labs, Inc.
// See LICENSE for copying information.

import { defineStore } from 'pinia';
import { ref } from 'vue';

import { AnalyticsHttpApi } from '@/api/analytics';
import { AnalyticsErrorEventSource, AnalyticsEvent } from '@/utils/constants/analyticsEventNames';

type Plausible = (event: 'pageview', data: { u: string; props: { source: string } }) => void;

export const useAnalyticsStore = defineStore('analytics', () => {
    const analytics: AnalyticsHttpApi = new AnalyticsHttpApi();

    const plausible = ref<Plausible>();

    async function loadPlausible(data: {
        domain: string;
        scriptURL: string;
    }) {
        try {
            await new Promise((resolve, reject) => {
                const head = document.head || document.getElementsByTagName('head')[0];
                const script = document.createElement('script');
                script.type = 'text/javascript';
                script.src = data.scriptURL;
                script.setAttribute('data-domain', data.domain);

                head.appendChild(script);
                script.onload = resolve;
                script.onerror = reject;
            });
            plausible.value = window['plausible'];
        } catch (_) { /*empty*/ }
    }

    function eventTriggered(eventName: AnalyticsEvent, props?: Map<string, string>): void {
        analytics.eventTriggered(eventName, props).catch(_ => { });
    }

    function linkEventTriggered(eventName: AnalyticsEvent, link: string): void {
        analytics.linkEventTriggered(eventName, link).catch(_ => { });
    }

    function pageVisit(pagePath: string, source: string): void {
        analytics.pageVisit(pagePath).catch(_ => { });

        if (!plausible.value) {
            return;
        }

        let url: string;
        if (pagePath.includes('http')) {
            // external link
            url = pagePath;
        } else {
            url = window.location.protocol + '//' + window.location.host + pagePath;
            if (window.location.search) {
                // remove sensitive query params
                const avoidKeys = ['token', 'email', 'inviter_email', 'projectID'];
                const filteredParams: string[] = [];
                const params = window.location.search.replace('?', '').split('&');
                for (const param of params) {
                    if (!avoidKeys.find((k) => param.includes(k))) {
                        filteredParams.push(param);
                    }
                }
                if (filteredParams.length > 0) url = url + '?' + filteredParams.join('&');
            }
        }

        plausible.value('pageview', { u: url, props: { source } });
    }

    function errorEventTriggered(source: AnalyticsErrorEventSource): void {
        analytics.errorEventTriggered(source).catch(_ => { });
    }

    return {
        loadPlausible,
        eventTriggered,
        errorEventTriggered,
        linkEventTriggered,
        pageVisit,
    };
});
