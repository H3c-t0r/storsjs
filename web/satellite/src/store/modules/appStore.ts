// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

import { reactive } from 'vue';
import { defineStore } from 'pinia';

import { OnboardingOS, PricingPlanInfo } from '@/types/common';
import { FetchState } from '@/utils/constants/fetchStateEnum';
import { ManageProjectPassphraseStep } from '@/types/managePassphrase';
import { FrontendConfig, FrontendConfigApi } from '@/types/config';
import { FrontendConfigHttpApi } from '@/api/config';

class ViewsState {
    public fetchState = FetchState.LOADING;
    public isSuccessfulPasswordResetShown = false;
    public hasJustLoggedIn = false;
    public onbAGStepBackRoute = '';
    public onbAPIKeyStepBackRoute = '';
    public onbCleanApiKey = '';
    public onbApiKey = '';
    public setDefaultPaymentMethodID = '';
    public deletePaymentMethodID = '';
    public onbSelectedOs: OnboardingOS | null = null;
    public selectedPricingPlan: PricingPlanInfo | null = null;
    public managePassphraseStep: ManageProjectPassphraseStep | undefined = undefined;
    public activeDropdown = 'none';
    // activeModal could be of VueConstructor type or Object (for composition api components).
    public activeModal: unknown | null = null;
    // this field is mainly used on the all projects dashboard as an exit condition
    // for when the dashboard opens the pricing plan and the pricing plan navigates back repeatedly.
    public hasShownPricingPlan = false;
    public error: ErrorPageState = new ErrorPageState();
    public isLargeUploadNotificationShown = true;
    public isLargeUploadWarningNotificationShown = false;
}

class ErrorPageState {
    constructor(
        public statusCode = 0,
        public fatal = false,
        public visible = false,
    ) {}
}

export class State {
    public viewsState: ViewsState = new ViewsState();
    public config: FrontendConfig = new FrontendConfig();
}

export const useAppStore = defineStore('app', () => {
    const state = reactive<State>(new State());

    const configApi: FrontendConfigApi = new FrontendConfigHttpApi();

    async function getConfig(): Promise<FrontendConfig> {
        const result = await configApi.get();

        state.config = result;

        return result;
    }

    function toggleActiveDropdown(dropdown: string): void {
        if (state.viewsState.activeDropdown === dropdown) {
            state.viewsState.activeDropdown = 'none';
            return;
        }

        state.viewsState.activeDropdown = dropdown;
    }

    function toggleSuccessfulPasswordReset(): void {
        if (!state.viewsState.isSuccessfulPasswordResetShown) {
            state.viewsState.activeDropdown = 'none';
        }

        state.viewsState.isSuccessfulPasswordResetShown = !state.viewsState.isSuccessfulPasswordResetShown;
    }

    function updateActiveModal(modal: unknown): void {
        // modal could be of VueConstructor type or Object (for composition api components).
        if (state.viewsState.activeModal === modal) {
            state.viewsState.activeModal = null;
            return;
        }
        state.viewsState.activeModal = modal;
    }

    function removeActiveModal(): void {
        state.viewsState.activeModal = null;
    }

    function toggleHasJustLoggedIn(hasJustLoggedIn: boolean | null = null): void {
        if (hasJustLoggedIn === null) {
            state.viewsState.hasJustLoggedIn = !state.viewsState.hasJustLoggedIn;
            return;
        }
        state.viewsState.hasJustLoggedIn = hasJustLoggedIn;
    }

    function changeState(newFetchState: FetchState): void {
        state.viewsState.fetchState = newFetchState;
    }

    function setOnboardingBackRoute(backRoute: string): void {
        state.viewsState.onbAGStepBackRoute = backRoute;
    }

    function setOnboardingAPIKeyStepBackRoute(backRoute: string): void {
        state.viewsState.onbAPIKeyStepBackRoute = backRoute;
    }

    function setOnboardingAPIKey(apiKey: string): void {
        state.viewsState.onbApiKey = apiKey;
    }

    function setOnboardingCleanAPIKey(apiKey: string): void {
        state.viewsState.onbCleanApiKey = apiKey;
    }

    function setOnboardingOS(os: OnboardingOS): void {
        state.viewsState.onbSelectedOs = os;
    }

    function setPricingPlan(plan: PricingPlanInfo): void {
        state.viewsState.selectedPricingPlan = plan;
    }

    function setManagePassphraseStep(step: ManageProjectPassphraseStep | undefined): void {
        state.viewsState.managePassphraseStep = step;
    }

    function setHasShownPricingPlan(value: boolean): void {
        state.viewsState.hasShownPricingPlan = value;
    }

    function setLargeUploadWarningNotification(value: boolean): void {
        state.viewsState.isLargeUploadWarningNotificationShown = value;
    }

    function setLargeUploadNotification(value: boolean): void {
        state.viewsState.isLargeUploadNotificationShown = value;
    }

    function closeDropdowns(): void {
        state.viewsState.activeDropdown = '';
    }

    function setErrorPage(statusCode: number, fatal = false): void {
        state.viewsState.error = new ErrorPageState(statusCode, fatal, true);
    }

    function removeErrorPage(): void {
        state.viewsState.error.visible = false;
    }

    function clear(): void {
        state.viewsState.activeModal = null;
        state.viewsState.isSuccessfulPasswordResetShown = false;
        state.viewsState.hasJustLoggedIn = false;
        state.viewsState.onbAGStepBackRoute = '';
        state.viewsState.onbAPIKeyStepBackRoute = '';
        state.viewsState.onbCleanApiKey = '';
        state.viewsState.onbApiKey = '';
        state.viewsState.setDefaultPaymentMethodID = '';
        state.viewsState.deletePaymentMethodID = '';
        state.viewsState.onbSelectedOs = null;
        state.viewsState.managePassphraseStep = undefined;
        state.viewsState.selectedPricingPlan = null;
        state.viewsState.hasShownPricingPlan = false;
        state.viewsState.activeDropdown = '';
        state.viewsState.error.visible = false;
    }

    return {
        state,
        getConfig,
        toggleActiveDropdown,
        toggleSuccessfulPasswordReset,
        updateActiveModal,
        removeActiveModal,
        toggleHasJustLoggedIn,
        changeState,
        setOnboardingBackRoute,
        setOnboardingAPIKeyStepBackRoute,
        setOnboardingAPIKey,
        setOnboardingCleanAPIKey,
        setOnboardingOS,
        setPricingPlan,
        setManagePassphraseStep,
        setHasShownPricingPlan,
        setLargeUploadWarningNotification,
        setLargeUploadNotification,
        closeDropdowns,
        setErrorPage,
        removeErrorPage,
        clear,
    };
});
