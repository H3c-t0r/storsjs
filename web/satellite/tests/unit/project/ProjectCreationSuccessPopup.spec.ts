// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import ProjectCreationSuccessPopup from '@/components/project/ProjectCreationSuccessPopup.vue';

import { appStateModule } from '@/store/modules/appState';
import { APP_STATE_ACTIONS } from '@/utils/constants/actionNames';
import { mount } from '@vue/test-utils';

const store = new Vuex.Store({ modules: { appStateModule }});

describe('ProjectCreationSuccessPopup.vue', () => {
    it('renders correctly', async (): Promise<void> => {
        await store.dispatch(APP_STATE_ACTIONS.TOGGLE_SUCCESSFUL_PROJECT_CREATION_POPUP);

        const wrapper = mount(ProjectCreationSuccessPopup, {
            store,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('closes correctly', async (): Promise<void> => {
        const wrapper = mount(ProjectCreationSuccessPopup, {
            store,
        });

        await wrapper.find('.project-creation-success-popup__close-cross-container').trigger('click');

        expect(wrapper).toMatchSnapshot();
    });
});
