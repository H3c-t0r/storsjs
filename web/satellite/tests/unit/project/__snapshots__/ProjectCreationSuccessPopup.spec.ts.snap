// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCreationSuccessPopup.vue closes correctly 1`] = ``;

exports[`ProjectCreationSuccessPopup.vue renders correctly 1`] = `
<div class="project-creation-success-popup-container">
  <div id="successfulProjectCreationPopup" class="project-creation-success-popup">
    <div class="project-creation-success-popup__info-panel-container"><svg width="123" height="123" viewBox="0 0 123 123" fill="none" xmlns="http://www.w3.org/2000/svg">
        <circle cx="61.5" cy="61.5" r="59.5" fill="white" stroke="#34BF89" stroke-width="4"></circle>
        <path d="M28.9292 66.1529C25.9126 63.1363 25.9126 58.2453 28.9292 55.2287C31.9459 52.212 36.8368 52.212 39.8535 55.2287L58.0606 73.4358C61.0773 76.4524 61.0773 81.3434 58.0606 84.3601C55.044 87.3767 50.153 87.3767 47.1363 84.3601L28.9292 66.1529Z" fill="#34BF89"></path>
        <path d="M58.0606 84.3601C55.044 87.3767 50.153 87.3767 47.1363 84.3601C44.1197 81.3434 44.1197 76.4524 47.1363 73.4358L79.9092 40.663C82.9258 37.6463 87.8168 37.6463 90.8334 40.663C93.8501 43.6796 93.8501 48.5706 90.8334 51.5872L58.0606 84.3601Z" fill="#34BF89"></path>
      </svg>
      <p class="project-creation-success-popup__info-panel-container__title">Project Created!</p>
    </div>
    <div class="project-creation-success-popup__form-container">
      <h2 class="project-creation-success-popup__form-container__main-label-text">Congrats!</h2>
      <p class="project-creation-success-popup__form-container__confirmation-text">You just created your project. Next, we recommend you create your first API Key for this project. API Keys allow developers to manage their projects and build applications on top of the Storj network through our
        <a href="https://documentation.tardigrade.io/api-reference/uplink-cli" target="_blank" class="project-creation-success-popup__form-container__confirmation-text__link">Uplink CLI.</a></p>
      <div class="project-creation-success-popup__form-container__button-container">
        <div class="container white" style="width: 214px; height: 50px;">
          <h1 class="label">I will do it later</h1>
        </div>
        <div class="container" style="width: 214px; height: 50px;">
          <h1 class="label">Create API Key</h1>
        </div>
      </div>
    </div>
    <div class="project-creation-success-popup__close-cross-container"><svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M15.7071 1.70711C16.0976 1.31658 16.0976 0.683417 15.7071 0.292893C15.3166 -0.0976311 14.6834 -0.0976311 14.2929 0.292893L15.7071 1.70711ZM0.292893 14.2929C-0.0976311 14.6834 -0.0976311 15.3166 0.292893 15.7071C0.683417 16.0976 1.31658 16.0976 1.70711 15.7071L0.292893 14.2929ZM1.70711 0.292893C1.31658 -0.0976311 0.683417 -0.0976311 0.292893 0.292893C-0.0976311 0.683417 -0.0976311 1.31658 0.292893 1.70711L1.70711 0.292893ZM14.2929 15.7071C14.6834 16.0976 15.3166 16.0976 15.7071 15.7071C16.0976 15.3166 16.0976 14.6834 15.7071 14.2929L14.2929 15.7071ZM14.2929 0.292893L0.292893 14.2929L1.70711 15.7071L15.7071 1.70711L14.2929 0.292893ZM0.292893 1.70711L14.2929 15.7071L15.7071 14.2929L1.70711 0.292893L0.292893 1.70711Z" fill="#384B65" class="close-cross-svg-path"></path>
      </svg></div>
  </div>
</div>
`;
